client/src/actions/index.js

```diff
@@ -79,26 +73,35 @@ return promise fetching layout we need
 function layoutFetchAndLoad(dispatch, schema) {
   const embeddings = schema?.schema?.layout?.obs ?? [];
   const embNames = embeddings.map((e) => e.name);
-  const baseURL = `${globals.API.prefix}${globals.API.version}layout/obs`;
 
   const plimit = new PromiseLimit(5);
   return Promise.all(
     embNames.map((e) =>
       plimit.add(() => {
-        const url = `${baseURL}?layout-name=${encodeURIComponent(e)}`;
-        return doBinaryRequest(url).then((buffer) =>
-          Universe.matrixFBSToDataframe(buffer)
-        );
+        return fetchBinary(`layout/obs?layout-name=${encodeURIComponent(e)}`)
+          .then(buffer => Universe.matrixFBSToDataframe(buffer));
       })
     )
-  ).then((dfs) => {
-    const df = Dataframe.Dataframe.empty().withColsFromAll(dfs);
+  ).then(dfs =>
     dispatch({
       type: "universe: column load success",
       dim: "obsLayout",
-      dataframe: df,
-    });
-  });
+      dataframe: Dataframe.Dataframe.empty().withColsFromAll(dfs)
```

Comment:
in some places (like this) you are removing blocks and intermediate variables (like `df`), and in others not (eg, line 55/56).  Should probably shoot for a single style.